addons/l10n_eu_service/models/res_company.py

```diff
@@ -0,0 +1,64 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, fields, models, _
+from .eu_tax_map import EU_TAX_MAP
+
+
+class Company(models.Model):
+    _inherit = 'res.company'
+
+    def _get_repartition_line_copy_values(self, original_rep_lines):
+        return [(0, 0, {
+            'factor_percent': line.factor_percent,
+            'repartition_type': line.repartition_type,
+            'account_id': line.repartition_type == 'tax' and line.account_id.id or None,
+            'company_id': line.company_id.id,
+            'sequence': line.sequence,
+        }) for line in original_rep_lines]
+
+    def map_eu_taxes(self):
+        eu_countries = self.env.ref('base.europe').country_ids
+        tax_group_id = self.env.ref('l10n_eu_service.tax_group_oss').id
+        for company in self:
+            tax_ids = self.env['account.tax'].search([('type_tax_use', '=', 'sale'),
+                ('amount_type', '=', 'percent'),
+                ('company_id','=', company.id)])
+            for domestic_tax in tax_ids:
+                for country in eu_countries - company.country_id:
+                    tax_amount = EU_TAX_MAP.get(company.country_id.code + str(domestic_tax.amount) + country.code, False)
+                    if tax_amount:
+                        tax_name =_('%(rate)s%% %(country)s %(label)s', rate=tax_amount, country=country.code, label=country.vat_label)
+                        foreign_tax = self.env['account.tax'].search([('name', '=', tax_name), ('company_id', '=', company.id)])
```

Comment:
Instead of relying on the name, you could keep a dictionary of everything you created so far in the refactored version of the loop I proposed above, just initializing it between the two loops (and so, reinitializing it for each country), containing {foreign_rate: created_tax}

Reply:
ok, done